src/components/WalletConnect/NanoContract/NanoContractMethodArgs.js

```diff
@@ -0,0 +1,141 @@
+/**
+ * Copyright (c) Hathor Labs and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import React, { useMemo } from 'react';
+import {
+  StyleSheet,
+  View,
+  Text,
+} from 'react-native';
+import { useDispatch, useSelector } from 'react-redux';
+import { t } from 'ttag';
+import { get } from 'lodash';
+import { COLORS } from '../../../styles/themes';
+import { commonStyles } from '../theme';
+import { onExceptionCaptured } from '../../../actions';
+import { NANOCONTRACT_BLUEPRINTINFO_STATUS as STATUS } from '../../../constants';
+
+/**
+ * Get method info from registered blueprint data.
+ *
+ * @param {{
+ *   data: Object;
+ * }} blueprint The blueprint info object
+ * @param {string} method The method name to get info from blueprint public methods
+ *
+ * @returns {Object}
+ */
+function getMethodInfoFromBlueprint(blueprint, method) {
+  return get(blueprint.data, `public_methods.${method}`, null);
+}
+
+/**
+ * Get the fallback entries for the method arguments.
+ *
+ * @param {string[]} args A list of argument value
+ *
+ * @returns {[argName: string, value: string][]}
+ *
+ * @example
+ * getFallbackArgEntries([...argValues])
+ * >>> [['Position 0', 'abc'], ['Position 1', '00'], ['Position 2', 123]]
+ */
+function getFallbackArgEntries(args) {
+  return args.map((arg, idx) => [t`Position ${idx}`, arg]);
+}
+
+/**
+ * It renders a list of method arguments for when the Nano Contract executes.
+ *
+ * @param {Object} props
+ * @param {string} props.blueprintId ID of blueprint.
+ * @param {string} props.method Method's name.
+ * @param {string[]} props.ncArgs A list of method's argument.
+ */
+export const NanoContractMethodArgs = ({ blueprintId, method, ncArgs }) => {
+  if (!ncArgs.length) {
+    return null;
+  }
+  const dispatch = useDispatch();
+
+  const blueprintInfo = useSelector((state) => state.nanoContract.blueprint[blueprintId]);
+  // It results a in a list of entries like:
+  // >>> [['oracle_script', 'abc'], ['token_uid', '00'], ['date_last_bet', 123]]
+  // or a fallback like:
+  // >>> [['Position 0', 'abc'], ['Position 1', '00'], ['Position 2', 123]]
+  const argEntries = useMemo(() => {
+    // Retrieve the fallback entries while blueprint info is being loading.
+    if (blueprintInfo == null || blueprintInfo.status === STATUS.LOADING) {
+      return getFallbackArgEntries(ncArgs);
+    }
```

Comment:
I don't think I like the fallback to be used while data is loading, only in case there's an error. I think I prefer a loading instead of having Position 0, then change it to the argument name.
What do you think?

Reply:
Oh! I've removed it and now the screen keeps loading until the download of blueprint information.